docs/sources/reference/stdlib/map.md

```diff
@@ -0,0 +1,35 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/stdlib/map/
+description: Learn about map functions
+aliases:
+  - ./concat/ # /docs/alloy/latest/reference/stdlib/concat/
+menuTitle: map
+title: map
+---
+
+# map
+
+The `map` namespace contains functions related to objects.
+
+## map.inner_join
+
+The `map.inner_join` function allows you to join two arrays containing maps if certain keys have matching values in both maps.
+It takes several inputs, in this order:
+
+1. An array of maps (objects) which will be joined. The keys of the map are strings. 
+   Its value could have any Alloy type such as a string, integer, map, or a capsule.
+2. An array of maps (objects) which will be joined. The keys of the map are strings. 
+   Its value could have any Alloy type such as a string, integer, map, or a capsule.
+3. An array containing strings. The strings are the keys whose value has to match for maps to be joined.
+   If the set of keys don't identify a map uniquely, the resulting output may contain more maps than the total sum of maps from both input arrays.
+4. (optional; default: `"update"`) A merge strategy. It describes which value will be used if there is Can be set to either of:
+   * `update`: If there is already a key with the same name in the first map, it will be updated with the value from the second map.
+   * `none`: If both maps have different values for the same key, no such key will be present in the output map.
+5. (optional; default: `[]`) An array containing strings. Only keys listed in the array will be present in the output map.
+6. (optional; default: `[]`) An array containing strings. Keyls listed in the array will not be present in the output map.
```

Comment:
I might remove those two parameters. I added them for two reasons:
* I thought it might make the function more ergonomic so that users don't have to use discovery.relabel every time, even if they want to do something basic.
* I thought it could be an optimisation to handle duplicate maps in a smart way. E.g. if the user only cares about adding 1 label, but their kubernetes SD gives them 20 labels, `map.inner_join` could remove the labels from the incoming map and then deduplicate the maps from the SD. 

But the more I think of this, the more it sounds like an overkill. Even if we do all this deduplication, there is still a chance that the keys used as condition of join are not unique, which could lead to too many output maps in the outgoing array. So it's probably best to let the users massage the data before the inner_join to make sure it's good.

Reply:
I removed those params